apps/beeai-ui/src/utils/feature-flags.ts

```diff
@@ -0,0 +1,32 @@
+/**
+ * Copyright 2025 © BeeAI a Series of LF Projects, LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { z } from 'zod';
+
+import { objectFromEntries } from './helpers';
+
+const featureNames = ['Variables', 'Providers'] as const;
+const featureFlagsDefaults = objectFromEntries(featureNames.map((key) => [key, false] as const));
+
+const featureFlagsSchema = z
+  .object(objectFromEntries(featureNames.map((key) => [key, z.boolean().optional().default(false)])))
+  .strict();
+
+export type FeatureFlags = z.infer<typeof featureFlagsSchema>;
+export type FeatureName = (typeof featureNames)[number];
+
+export function parseFeatureFlags(data?: string) {
+  if (data) {
```

Comment:
I personally don't like the nesting because of `data`. We could either change it to a guard clause, or remove completely.